R/prepare_locs.R

```diff
@@ -6,7 +6,9 @@ prepare_locs <- function(DT, id, datetime, tz, x, y, split_by) {
 
 	DT[, (datetime) := parse_date(datetime_char, default_tz = tz),
 		 env = list(datetime_char = datetime)]
-
+	# changing IDate format to a different variable to avoid iteration = "group" error later on
+	DT[, idate := as.Date(idate)]
```

Comment:
I think we need a more general solution here, because it will depend on the input data for folks using the template whether or not they have IDate columns in their data

What about this more general solution that casts any IDate columns (if any) as Date?

```r
# Cast any IDate columns as Date
is_idate <- vapply(DT, function(x) 'IDate' %in% class(x), FUN.VALUE = TRUE)
DT[, (names(is_idate)) := lapply(.SD, as.Date), .SDcols = names(is_idate)]
```